release.sh

```diff
@@ -5,6 +5,31 @@ cd $DIR
 
 set -e
 
+git remote update
+
+remote=$(git rev-parse "@{u}") # "@{u}" gets the current upstream branch
+local=$(git rev-parse @) # '@' gets the current local branch 
+
+# check if local commit syncs with remote 
+if [ "$remote" != "$local" ]
+    then
+        printf "%s" "Error: local commit does not match remote. Exiting release script." $'\n'
+        exit 1
+fi 
+
+# remove any excess brackets, space/tab characters and 'origin' branch, then sort the tags
+remoteTags=$(git ls-remote --tags origin | grep -v '{}' | sort | tr -d [[:blank:]])
+localTags=$(git show-ref --tags | grep -v '{}' | grep -v 'origin'| sort | tr -d [[:blank:]])
+
+# check if sorted local tags match remote tags
+if ! diff -q <(printf "%s" "$remoteTags") <(printf "%s" "$localTags") >/dev/null 2>&1
+    then 
+        printf "%s" "Error: local tags do not match remote. Exiting release script." $'\n'
```

Comment:
You can just use `echo` for this. The `printf` is mostly just for untrusted input.